clang/lib/StaticAnalyzer/Checkers/WebKit/ASTUtils.cpp

```diff
@@ -70,6 +70,8 @@ bool tryToFindPtrOrigin(
           if (isCtorOfSafePtr(ConversionFunc))
             return callback(E, true);
         }
+        if (isa<CXXFunctionalCastExpr>(E) && isSafePtrType(cast->getType()))
+          return callback(E, true);
```

Comment:
Is this case already handled for parameters and locals? Like are there instances of a constructing a Ref from another Ref parameter?

Reply:
We typically don't pass arguments as RefPtr / Ref except a few cases where we pass in as RefPtr&& / Ref&&. I guess we can add a test for it though.